src/main/java/uk/gov/hmcts/darts/task/service/impl/ManualDeletionProcessorImpl.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.darts.task.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.audio.deleter.impl.inbound.ExternalInboundDataStoreDeleter;
+import uk.gov.hmcts.darts.audio.deleter.impl.unstructured.ExternalUnstructuredDataStoreDeleter;
+import uk.gov.hmcts.darts.common.entity.ExternalObjectDirectoryEntity;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.entity.ObjectAdminActionEntity;
+import uk.gov.hmcts.darts.common.entity.TranscriptionDocumentEntity;
+import uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.MediaRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectAdminActionRepository;
+import uk.gov.hmcts.darts.common.repository.TranscriptionDocumentRepository;
+import uk.gov.hmcts.darts.task.service.ManualDeletionProcessor;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.List;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ManualDeletionProcessorImpl implements ManualDeletionProcessor {
+
+    @Value("${darts.manual-deletion.grace-period:24h}")
+    private String gracePeriod;
+
+    private final ObjectAdminActionRepository objectAdminActionRepository;
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final MediaRepository mediaRepository;
+    private final TranscriptionDocumentRepository transcriptionDocumentRepository;
+    private final ExternalInboundDataStoreDeleter inboundDeleter;
+    private final ExternalUnstructuredDataStoreDeleter unstructuredDeleter;
+
+    @Override
+    public void process() {
+        OffsetDateTime deletionThreshold = getDeletionThreshold();
+        List<ObjectAdminActionEntity> actionsToDelete = objectAdminActionRepository.findFilesForManualDeletion(deletionThreshold);
+
+        for (ObjectAdminActionEntity objectAdminAction : actionsToDelete) {
+            if (isMediaNotDeleted(objectAdminAction)) {
+                deleteMedia(objectAdminAction.getMedia());
+            } else if (isTranscriptionNotDeleted(objectAdminAction)) {
+                deleteTranscriptionDocument(objectAdminAction.getTranscriptionDocument());
+            }
+        }
+
+    }
+
+    private void deleteMedia(MediaEntity mediaEntity) {
+
+        externalObjectDirectoryRepository.findByMediaStatusAndLocation(mediaEntity.getId())
+            .stream()
```

Comment:
streams have a number of drawbacks, including less performant than a standard for-loop, and they make it difficult to debug, we try to limit the amount of "work" a thread does due to this, so it may be an idea to use a for-loop instead here.

Reply:
replaced with for loop